quinn-proto/src/connection/mod.rs

```diff
@@ -233,6 +233,74 @@ pub struct Connection {
     stats: ConnectionStats,
     /// QUIC version used for the connection.
     version: u32,
+
+    /// qlog streamer to ouput events as JSON text sequences
+    #[cfg(feature = "qlog")]
+    qlog_streamer: Option<qlog::streamer::QlogStreamer>,
+
+    /// Snapshot of the qlog recovery metrics
+    #[cfg(feature = "qlog")]
+    recovery_metrics: CongestionMetrics,
+}
+
+#[cfg(feature = "qlog")]
+#[derive(Default, PartialEq)]
+pub struct CongestionMetrics {
```

Comment:
Should this be `#[non_exhaustive]`? That would allow us to add new fields in the future without breaking congestion controller impls.